aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -386,51 +386,108 @@ inline Tensor reduce_multi_associative(const Tensor &self, IntList dims_, bool k
     return self;
   }
   int64_t ndims = self.dim();
-  auto reduce_dims = dim_list_to_bitset(dims_, ndims);
-  Tensor result = self;
-  for (int64_t dim = ndims-1; dim >= 0; dim--) {
-    if (reduce_dims[dim])
-      result = reduce_1(result, dim, keepdim);
+  auto reduced_size = self.sizes().vec();
```

Comment:
I don't actually know if this is the case or not, but does this do the optimization where if two dimensions are contiguous, they are collapsed into a single dimension and reduced all at once?